subprojects/core/src/main/java/org/gradle/internal/service/scopes/VirtualFileSystemServices.java

```diff
@@ -192,7 +192,7 @@ WatchingAwareVirtualFileSystem createVirtualFileSystem(
                 .orElse(new NonWatchingVirtualFileSystem(delegate));
             listenerManager.addListener(new VirtualFileSystemBuildLifecycleListener(
                 watchingAwareVirtualFileSystem,
-                startParameter -> isRetentionEnabled(startParameter.getSystemPropertiesArgs()),
+                StartParameter::isWatchFileSystem,
                 startParameter -> isSystemPropertyEnabled(VFS_DROP_PROPERTY, startParameter.getSystemPropertiesArgs()),
                 startParameter -> getSystemProperty(VFS_CHANGES_SINCE_LAST_BUILD_PROPERTY, startParameter.getSystemPropertiesArgs())
             ));
```

Comment:
I think we should drop `VFS_CHANGES_SINCE_LAST_BUILD_PROPERTY` now. It shouldn't be used by performance tests either.

Reply:
We should also retire `VFS_RETENTION_ENABLED_PROPERTY` and warn/fail if it's used, so as not to silently do the wrong thing in performance tests.

Reply:
I guess we should keep `VFS_DROP_PROPERTY`.

Reply:
I think we also should remove `VFS_PARTIAL_INVALIDATION_ENABLED_PROPERTY`, since we won't be enabling partial invalidation separately from watching the file system. WDYT?

Reply:
We could, but I'd be happier to remove it once we don't have an `if` around that feature. I'd leave the current switch in place until now.

Reply:
I removed `VFS_CHANGES_SINCE_LAST_BUILD_PROPERTY` and the first class support for configuring `VFS_PARTIAL_INVALIDATION_ENABLED_PROPERTY` in the executor. We are not using the support, so I don't see why we should keep it.